src/include/binder/expression/node_rel_expression.h

```diff
@@ -0,0 +1,35 @@
+#pragma once
+
+#include "expression.h"
+
+namespace kuzu {
+namespace binder {
+
+class NodeOrRelExpression : public Expression {
+public:
+    NodeOrRelExpression(
+        DataTypeID dataTypeID, const string& uniqueName, vector<table_id_t> tableIDs)
+        : Expression{VARIABLE, dataTypeID, uniqueName}, tableIDs{std::move(tableIDs)} {}
+
+    inline void addTableIDs(const vector<table_id_t>& tableIDsToAdd) {
+        auto tableIDsMap = unordered_set<table_id_t>(tableIDs.begin(), tableIDs.end());
+        for (auto tableID : tableIDsToAdd) {
+            if (!tableIDsMap.contains(tableID)) {
+                tableIDs.push_back(tableID);
+            }
+        }
+    }
+    inline bool isMultiLabeled() const { return tableIDs.size() > 1; }
+    inline vector<table_id_t> getTableIDs() const { return tableIDs; }
+    // We don't support multi-labeled update
```

Comment:
Remove this comment. This is incorrect, we will have this API eventually because we have fast-path code for single labeled query